api/src/main/java/org/openmrs/module/kenyaemr/reporting/cohort/definition/evaluator/ARTRegisterCohortDefinitionEvaluator.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.kenyaemr.reporting.cohort.definition.evaluator;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Cohort;
+import org.openmrs.annotation.Handler;
+import org.openmrs.module.kenyaemr.reporting.cohort.definition.ARTRegisterCohortDefinition;
+import org.openmrs.module.kenyaemr.reporting.cohort.definition.IPTRegisterCohortDefinition;
+import org.openmrs.module.reporting.cohort.EvaluatedCohort;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.evaluator.CohortDefinitionEvaluator;
+import org.openmrs.module.reporting.common.ObjectUtil;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.EvaluationException;
+import org.openmrs.module.reporting.evaluation.querybuilder.SqlQueryBuilder;
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+
+/**
+ * Evaluator for ART Register Cohort Evaluator
+ */
+@Handler(supports = {ARTRegisterCohortDefinition.class})
+public class ARTRegisterCohortDefinitionEvaluator implements CohortDefinitionEvaluator {
+
+    private final Log log = LogFactory.getLog(this.getClass());
+	@Autowired
+	EvaluationService evaluationService;
+
+    @Override
+    public EvaluatedCohort evaluate(CohortDefinition cohortDefinition, EvaluationContext context) throws EvaluationException {
+
+		ARTRegisterCohortDefinition definition = (ARTRegisterCohortDefinition) cohortDefinition;
+
+        if (definition == null)
+            return null;
+
+		Cohort newCohort = new Cohort();
+
+		context = ObjectUtil.nvl(context, new EvaluationContext());
+
+		String qry = "select patient_id\n" +
```

Comment:
@patryllus i doubt if this query will return the expected list of patients for the cohort register. I shared with you a query and am wondering where that went